projects/common/src/time/page-time-range-preference.service.ts

```diff
@@ -62,11 +83,8 @@ export class PageTimeRangePreferenceService {
       .pipe(shareReplay(1));
   }
 
-  private getDefaultPageTimeRange(path: string): TimeRange | undefined {
-    const defaultTimeRange = this.navigationService.getRouteConfig([path], this.navigationService.rootRoute())?.data
-      ?.defaultTimeRange;
-
-    return defaultTimeRange;
+  private getDefaultPageTimeRange(): TimeRange | undefined {
+    return this.navigationService.getCurrentActivatedRoute().snapshot.data?.defaultTimeRange;
```

Comment:
Just trying to understand how this works - if I click on A on the left nav, and it routes be to A/B, we need the time range to be defined on B right? But in the dictionary, if I changed it, it would be stored under A? If so, I think that's confusing. If we think A is the important bit there, then the time range being defined there would make sense too, so if I change the default routing it doesn't touch the time range. To do that, you might need to change this to walk up the tree looking.

Reply:
Yes that is how it works. @anandtiwary and I briefly looked into traversing the routes to find the correct activated route but it's a little tricky. Because all the navItem time ranges are resolved on initial load, you can't use ActivatedRoute.  That's why before i was using `navigationService.getRouteConfig` - however, this method does not traverse lazy loaded child routes, so we can't use it here either.  

One solution would be to keep it as-is with the A/B functionality you described above, and to create a future work item out of this.  Another option is to change the timing of when the navItems are decorated to being on click, but that might be a timely effort, requiring a rework in the nav-list component.

Reply:
> looked into traversing the routes to find the correct activated route but it's a little tricky

We do have this tree walk in a number of places today. For example the feature guard I believe starts from the activated route, and walks up the tree checking the config for each parent route.

> Because all the navItem time ranges are resolved on initial load, you can't use ActivatedRoute. That's why before i was using navigationService.getRouteConfig

Not sure why that would differ though - the activated route just gives you another way to get to the route config - either way it's resolved at load time, which is a very good point. We need to rebuild relative time ranges at the point in time where they take effect (will take a look back at the PR with that in mind).

> One solution would be to keep it as-is with the A/B functionality you described above, and to create a future work item out of this. Another option is to change the timing of when the navItems are decorated to being on click, but that might be a timely effort, requiring a rework in the nav-list component.

Don't think the latter is either worth it or would work. I think I would go for putting the time range on the root route's config, and changing the lookup to walk up the tree like we do in FF, but up to you if you feel it makes sense to do that now or later.